src/Middleware/Diagnostics/src/StatusCodePage/StatusCodePagesExtensions.cs

```diff
@@ -160,6 +162,47 @@ public static IApplicationBuilder UseStatusCodePagesWithReExecute(
         return app.UseStatusCodePages(CreateHandler(pathFormat, queryFormat));
     }
 
+    /// <summary>
+    /// Adds a StatusCodePages middleware to the pipeline. Specifies that the response body should be generated by
+    /// re-executing the request pipeline using an alternate path. This path may contain a '{0}' placeholder of the status code.
+    /// </summary>
+    /// <param name="app"></param>
+    /// <param name="pathFormat"></param>
+    /// <param name="createScopeForErrors">Whether or not to create a new <see cref="IServiceProvider"/> scope.</param>
+    /// <param name="queryFormat"></param>
+    /// <returns></returns>
+    [SuppressMessage("ApiDesign", "RS0026:Do not add multiple overloads with optional parameters", Justification = "Required to maintain compatibility")]
+    public static IApplicationBuilder UseStatusCodePagesWithReExecute(
+        this IApplicationBuilder app,
+        string pathFormat,
+        bool createScopeForErrors,
```

Comment:
I'm not sure about it, so take it, leave it (or wait for API review) but it's a bit strange that pathFormat and queryFormat are separated by a completely unrelated parameter. It might make sense to either move `createScopeForErrors` before `pathFormat` or after `queryFormat` and give it a `false` default value.

Reply:
I understand and agree.
I will:
1) reorder the args, it really looks strange
2) check in the 1st line check for `createScopeForErrors=false` and trigger the old `UseStatusCodePagesWithReExecute`. Otherwise, we have 2 methods that work a bit different, even though the params passed are same.